pinot-common/src/main/java/org/apache/pinot/filesystem/PinotFSFactory.java

```diff
@@ -43,6 +43,18 @@ private PinotFSFactory() {
 
   }
 
+  public static void register(String scheme, String fsClassName, Configuration configuration) {
```

Comment:
Can you provide examples of what are the variables that are expected to be set in Configuration? Or, is the whole thing totally FS dependent? (e.g. do we expect basedir to be set there?)

Reply:
only properties specific to FS.

Reply:
Then it should  be Properties. PinotFS implementations may have mulitple levels of configs.

Reply:
This method extracts the logic of creating a FileSystem, Configuration is what used to init PinotFSFactory. And fsConfig.subset(key) is used as the  Configuration to init internal PinotFS